_posts/2018-01-01-futures-0.3.0-alpha.3.md

```diff
@@ -0,0 +1,121 @@
+---
+layout: post
+title:  "Futures 0.3.0-alpha.3"
+author: "Josef Brandl"
+author_github: "MajorBreakfast"
+date:   2018-01-01
+categories: blog
+---
+
+`0.3.0-alpha.3` requires a recent nightly (2018-08-xx or newer):
+```sh
+$ rustup update
+```
+
+## Compatibility layer
+
+A compatibility layer between 0.3 an 0.1 was developped. It is now possible to convert an 0.3 future into an 0.1 future and vice versa. Similar conversions for streams and sinks are also supported. Additionally it is now possible to run 0.3 futures and async functions on Tokio's executor. We have a dedicated blog post coming up that explains this in more detail.
+
+## Spawning
+
+### `Spawn` trait
+
+The `std::task::Executor` trait is now called `Spawn`. The purpose of the trait remains unchanged - it is to provide a mechanism for spawning tasks. We noticed that in many implementations the responsibilty of spawning tasks is handled by a type that merely sends the task to an associated executor, not by the task executor type itself. Furthermore, it became clear that the former trait name "`Executor`" didn't properly capture the essence of what it means to be an executor. Fundamentally, a task executor is a type that manages the execution of tasks. There exists no fixed set of methods that a trait could define. Instead, each task executor type offers different methods that make sense for its design goals. This is why we chose to rename the trait to `Spawn` to reflect that it merely defines a method for spawning tasks and makes no specifications about how to run them.
+
+As a consequence of this renaming, some functions on `Context` were adjusted:
+```rust
+impl<'a> Context<'a> {
+    pub fn new(local_waker: &'a LocalWaker, spawner: &'a mut dyn Spawn) -> Context<'a> { ... }
+
+    pub fn spawner(&mut self) -> &mut dyn Spawn { ... }
+
+    pub fn with_spawner<'b, Sp: Spawn>(&'b mut self, spawner: &'b mut Sp) -> Context<'b>
+
+    ...
+}
+```
+### Spawn macros
```

Comment:
Should have an empty line above a heading (currently breaks github's syntax highlighting, but may work depending on the parser, nicer with the space when reading the source anyway)